src/App.tsx

```diff
@@ -1,10 +1,75 @@
+/* eslint-disable max-len */
+/* eslint-disable react/jsx-no-bind */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
+
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { SelectedCategory } from './types/SelectedCategory';
+import { ErrorMessage } from './types/ErrorMessage';
+
+import * as todoService from './api/todos';
+
+import { USER_ID } from './utils/constants';
+
+import { ToggleAll } from './component/ToggleAll';
+import { TodoInList } from './component/TodoInList';
+import { NewTodo } from './component/NewTodo';
+import { Filter } from './component/Filter';
+import { ClearCompleted } from './component/ClearCompleted';
+import { ErrorNotification } from './component/ErrorNotification';
+
+function filter(todos: Todo[], category: SelectedCategory) {
+  switch (category) {
+    case SelectedCategory.Active:
+      return [...todos].filter(todo => !todo.completed);
+
+    case SelectedCategory.Completed:
+      return [...todos].filter(todo => todo.completed);
+
+    default:
+      return [...todos];
+  }
+}
+
+const showError = (
+  message: ErrorMessage,
+  set: (error: ErrorMessage | null) => void,
+) => {
+  set(message);
+
+  setTimeout(() => {
+    set(null);
+  }, 3000);
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [category, setCategory] = useState<SelectedCategory>(SelectedCategory.All);
+  const [error, setError] = useState<ErrorMessage | null>(null);
+
+  useEffect(() => {
+    todoService
+      .getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => showError(ErrorMessage.Load, setError));
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return filter(todos, category);
+  }, [todos, category]);
+
+  const countActiveTodos = useMemo(() => {
```

Comment:
FYI
Try to use proper vars names) since `countActiveTodos` is a variable name, not a method it is better to name it as a noun like `activeTodosCount`. and if it was the method, for getting count it could be `getActiveTodosCount`, and eventually for methods that do something but not return we would use `countActiveTodos` the same as you used for `showError`

This is not a mistake though, but just keep in mind for the future)